### ISSUE TITLE
Global Signal 

### ISSUE THREAD

**Issue by vcolton123** [poldracklab/fmriprep#1015]
Could someone explain why I see negative values in the global signal output? If it is an average of the mask I expected to see positive values. 

Thanks!

**Comment by effigies** [comment]
Can you verify that your BOLD series has no negative values in it? If you're able, could you share a subject this occurs on on [OpenNeuro](https://openneuro.org/)?

**Comment by vcolton123** [comment]
This is exactly how it looks like: 
![screen shot 2018-03-01 at 2 05 28 pm](https://user-images.githubusercontent.com/36967998/36864122-b9303c06-1d59-11e8-9b9c-d237c74327eb.png)

**Comment by effigies** [comment]
Sorry, by BOLD series, I mean the functional NIFTI file. It should be in your BIDS directory as `sub-<subject_label>/func/sub-<subject_label>_task-<task_label>_bold.nii.gz` or similar.

**Comment by oesteban** [comment]
Since FMRIPREP uses the Lanczos interpolation, it is possible to see small negative values after resampling.

**Comment by effigies** [comment]
Please reopen if you have further questions.

**Comment by chrisfilo** [comment]
I don't think this is caused by interpolation. Could replicate it on other data
![image](https://user-images.githubusercontent.com/238759/36873859-ab6b1172-1d5e-11e8-95e2-623a729a9e2b.png)

https://openneuro.org/datasets/ds000007/versions/00001?app=FMRIPREP&version=50&job=5a808d16831753000151c638

**Comment by lenapl** [comment]
Folks - any  clarification on what is really being computed in the GS column? seems to be that it is unlikely to be the arithmetic mean of signals from all grey matter voxels!

**Comment by chrisfilo** [comment]
Thanks.

The negative voxels seem to be mostly outside of the brain - not sure the negative values for the Global Signal (which should be mean within brain mask voxels) could be explained by this.

**Comment by chrisfilo** [comment]
Well - even if they are within the mask they would need to amount to a large change.

**Comment by effigies** [comment]
Sorry, to be clear, that bottom image is masked before I filter for negative values. So these are within the mask. I'll let you know what else I find out.

**Comment by effigies** [comment]
But yes, you're right, if I directly take the mean, I get positive values. However, I think that identifies the issue. The `GlobalSignal` column has a linear trend of `1378.16 - 0.14992506*t` removed.

**Comment by effigies** [comment]
Oh, we use `detrend=True`. Should we remove this?

https://github.com/poldracklab/fmriprep/blob/83c4f1e0682b788cc1bac7831b218d5bf322821f/fmriprep/workflows/bold/confounds.py#L165-L167

**Comment by oesteban** [comment]
I guess that would actually remove the global signal, wouldn't it?

I think that `detrend=True` should definitely go away.

**Comment by effigies** [comment]
I agree that this doesn't seem to make much sense here. And just in general, I think it should be the responsibility of the model fitter to perform any manipulation on the confounds.

However, are we at all concerned by the addition of negative values to the resampled BOLD series?

**Comment by chrisfilo** [comment]
Depending on how detrending was implemented (before or after taking the average? which order trends were removed?) it would have different effects. Let's do a simple mean. **This applies to other region based regressors.** I agree that if someone wants to process the regressors they should do it on their own.

As for negative values - in principle no, since they could come from interpolation. However, in this case the negative values were an indication of an issue with how we calculate things.

**Comment by effigies** [comment]
I think the only region-based regressors we have are `WhiteMatter`, `CSF` and `GlobalSignal`, all of which are calculated in this node. So I think a one line fix will resolve the issue.

**Comment by lenapl** [comment]
thanks for looking into this, much appreciated
detrending is done across time, but GS values must be available for each time point- am I missing something?

**Comment by effigies** [comment]
What's being done is taking the arithmetic mean of all in-brain (gray + white + CSF) voxels. A simple linear detrend is being performed on the resulting 1D time series.

If you're removing linear trends in your GLM, as well as using the `GlobalSignal` column as a nuisance regressor, then there should be no change in results. The difference will appear in the beta values of your nuisance regressors, but that won't cause any problems. There might be very minor effects if you're high-pass filtering instead of detrending, but these should be on the order of rounding errors.

**Comment by lenapl** [comment]
thanks! I (and perhaps some others) would be interested in the actual GS value rather than using it as a regressor.  Absolute values will be very helpful, thanks

**Comment by effigies** [comment]
#1058 will move to that. It should make things less confusing, and make that step very slightly more efficient.
